### PR TITLE
Test panel classes

### PR DESCRIPTION
**Pull request by jaib1**
I've tested out these branches on the ZRigs to make sure mc experiments still run well with the changes I've made. Besides the changes to creating classes for running test experiments, I've also made a few other documentation/update changes. The small changes to files are typically variable name changes or documentation notes.

### CODE REVIEW

File: +dat/newExp.m [hunk 1]
```diff
@@ -60,10 +60,10 @@
 
 % if the parameters had an experiment definition function, save a copy in
 % the experiment's folder
-if isfield(expParams, 'defFunction')
-  assert(file.exists(expParams.defFunction),...
-    'Experiment definition function does not exist: %s', expParams.defFunction);
-  assert(all(cellfun(@(p)copyfile(expParams.defFunction, p),...
+if isfield(expParams, 'expDefFunction')
```
[the remark this answers is not in the document]
Comment (jaib1) [reply +dat/newExp.m]: don't feel super strongly, but think 'expDefFunction' is clearer. where/how would this break older experiments?

[the remark this answers is not in the document]
Comment (k1o0) [reply +dat/newExp.m]: All saved parameters and last used parameters use 'defFunction'.  You wouldn't be able to run these without renaming the fieldname.

File: +exp/SignalsExp.m [hunk 2]
```diff
@@ -192,8 +192,9 @@
         onValue(obj.Events.expStop, @(~)quit(obj));];
 %         obj.Events.trialNum.onValue(fun.partial(@fprintf, 'trial %i started\n'))];
       % initialise the parameter signals
-      globalPars.post(rmfield(globalStruct, 'defFunction'));
-      allCondPars.post(allCondStruct);
+      %globalPars.post(rmfield(globalStruct, 'expDefFunction'));
```
[the remark this answers is not in the document]
Comment (jaib1) [reply +exp/SignalsExp.m]: thought it's nice to see in the parameters panel? doesn't really cost anything?

File: +exp/runTrials.m [hunk 3]
```diff
@@ -6,16 +6,16 @@
   % parsData is number of trials, so turn number of trials into struct
   % array with no fields but 'parsData' elements
   globalStruct = struct;
-  allCondStruct = rmfield(struct('dummy', cell(1, parsData)), 'dummy');
+  condStruct = rmfield(struct('dummy', cell(1, parsData)), 'dummy');
```
[the remark this answers is not in the document]
Comment (jaib1) [reply +exp/runTrials.m]: ok, will keep it in for now and save it for the 'cleanup' branch

File: +dat/newExp.m (hunk 1 above)
[the remark this answers is not in the document]
Comment (jaib1) [reply +dat/newExp.m]: I think I made the necessary corrections for this, and have just updated 'eui.ExpPanel' to take this into account too. Just ran test cases on the zRigs

**Comment by jaib1** [comment]
I've made some changes, (including taking 'cprintf' out - not worth the citation =D ) but left the variable name changes 'expDefFunction' and 'condStruct' in. Checked to make sure it's running fine on the rigs - including with loading old parameter sets that use 'defFunction'. Let me know if you want to talk any more about it, otherwise all set to merge into 'dev' (which will have to be done in the command line because of the tiny conflicts in 'eui.MControl' and 'hw.devices' that should be incorporated from this branch).

**Comment by jaib1** [comment]
more changes. if looks good now (with access attributes set properly), let me know. I can do the merge in command line if you want me to.

**Comment by jaib1** [comment]
> I can do the merge in command line if you want me to (and squash and rebase before to put this branch all into one commit)

I've already done this on my local, because I've created a new branch for testing I wanted to branch directly off from this merger. if it looks good to you, I'll just push the 'dev' from my local to both **Rigbox** and **signals**

File: +exp/SignalsExp.m (hunk 2 above)
[the remark this answers is not in the document]
Comment (k1o0) [reply +exp/SignalsExp.m]: I think instead we could rename the 'custom' label or something? This just adds another param that's saved n times in the block (where n is the number of trials) and also sent to mc with the live updates.  Also note that exp.test sets the exp def as the figure title.

**Comment by jaib1** [comment]
re-done again, let me know if you'd like me to do the merge into dev on my local and push

**Pull request merged**